src/kademlia/get_item.cpp

```diff
@@ -170,92 +162,28 @@ bool get_item::invoke(observer_ptr o)
 
 void get_item::done()
 {
+	// only for put_immutable_item.
+	if (!m_data_callback) return find_data::done();
```

Comment:
Yes and Yes. 
For immutable_put, we only need nodes_callback then we can write item to them.
For mutable_put, we need data_callback to update the sequence number, so it must be set.